setup.py

```diff
@@ -48,25 +70,66 @@ class PactPythonInstallCommand(install):
 
     Installs the Python package and unpacks the platform appropriate version
     of the Ruby mock service and provider verifier.
+
+    User Options:
+        --bin-path  An absolute folder path containing predownloaded pact binaries
+                    that should be used instead of fetching from the internet.
     """
 
+    user_options = install.user_options + [('bin-path=', None, None)]
+
+    def initialize_options(self):
+        """Load our preconfigured options"""
+        install.initialize_options(self)
+        self.bin_path = None
+
+    def finalize_options(self):
+        """Load provided CLI arguments into our options"""
+        install.finalize_options(self)
+
     def run(self):
         """Install python binary."""
         install.run(self)
-        bin_path = os.path.join(self.install_lib, 'pact', 'bin')
-        os.mkdir(bin_path)
-        install_ruby_app(bin_path)
+        package_bin_path = os.path.join(self.install_lib, 'pact', 'bin')
+        if not os.path.exists(package_bin_path):
+            os.mkdir(package_bin_path)
+        install_ruby_app(package_bin_path, self.bin_path)
 
 
-def install_ruby_app(bin_path):
+def install_ruby_app(package_bin_path, download_bin_path):
     """
-    Download a Ruby application and install it for use.
+    Installs the ruby standalone application for this OS.
 
-    :param bin_path: The path where binaries should be installed.
+    :param package_bin_path: The path where we want our pact binaries unarchived.
+    :param download_bin_path: An optional path containing pre-downloaded pact binaries.
+    """
+
+    binary = ruby_app_binary()
+    if download_bin_path is None:
+        download_bin_path = package_bin_path
+
+    path = os.path.join(download_bin_path, binary['filename'])
+
+    if os.path.isfile(path) is True:
+        extract_ruby_app_binary(download_bin_path, package_bin_path, binary['filename'])
+    else:
+        if download_bin_path is not None:
+            if os.path.isfile(path) is not True:
+                raise RuntimeError('Could not find {} binary.'.format(path))
+            extract_ruby_app_binary(download_bin_path, package_bin_path, binary['filename'])
```

Comment:
Is this line reachable?

Reply:
I think this code will be hit if the user supplies a `bin-path` (i.e. `download_bin_path`) CLI argument and it contains the necessary standalone binaries

Reply:
@fresto32 please note that in line 108 we check the opposite condition, so therefore `download_bin_path is not None` will never be false. I think the current behavior is broken, when I try to install, the binary is not downloaded currently. It worked for me after I removed the if and left:
```
download_ruby_app_binary(download_bin_path, binary['filename'], binary['suffix'])
extract_ruby_app_binary(download_bin_path, package_bin_path, binary['filename'])
```